lib/rules/no-unused-prop-types.js

```diff
@@ -668,10 +602,12 @@ module.exports = {
               )
             );
             // let {firstname} = props
-            const genericDestructuring = isPropAttributeName(node) && (
-              utils.getParentStatelessComponent() ||
-              isInLifeCycleMethod(node)
-            );
+            const genericDestructuring =
+              isPropAttributeName(node) && (
+                utils.getParentStatelessComponent() ||
+                astUtil.inConstructor(context) ||
```

Comment:
which test case exposes this missing inConstructor check?

Reply:
Note that the old `isInLifeCycleMethod` function already was checking whether we are in constructor: `node.kind === 'constructor'`. Now it is split into `inConstructor` and `inLifecycleMethod` so the code is the same between this rule and `prop-types`.

In this case, the following two tests will fail when removing this line:

* https://github.com/yannickcr/eslint-plugin-react/blob/master/tests/lib/rules/no-unused-prop-types.js#L2037
* https://github.com/yannickcr/eslint-plugin-react/blob/master/tests/lib/rules/no-unused-prop-types.js#L1213